src/http/server/context.cr

```diff
@@ -7,8 +7,11 @@ class HTTP::Server
     # The `HTTP::Server::Response` to configure and write to.
     getter response : Response
 
+    # An optional `Socket` which holds the client connection of this context.
+    getter socket : Socket?
```

Comment:
I'm not sure about this.

Reply:
Same here. Hijacking the IO must be achieved within a request handler or response output (see WebSocket handler).

Reply:
The websocket handler really hijacks the IO using `upgrade`. The problem I'm trying to solve is not about taking over the IO handling (including responsibility for calling `#close`). It's rather to get information about the connection, such as `local_address`. Don't know about other things... `remote_address` could be useful.
So maybe we could just expose those in `Response`?

Reply:
I'd prefer to expose certain metadata regarding the socket, instead of an actual socket you can perform IO on.